platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventUtils.java

```diff
@@ -54,4 +56,38 @@ public static Instant calculateNewEventCreationTime(
             return now;
         }
     }
+
+    /**
+     * Returns the timestamp of the transaction with given index in this event
+     *
+     * @param transactionIndex index of the transaction in this event
+     * @return timestamp of the given index transaction
+     */
+    public static @NonNull Instant getTransactionTime(@NonNull final GossipEvent event, final int transactionIndex) {
+        if (event.getConsensusTimestamp() == null) {
+            throw new IllegalArgumentException("Event is not a consensus event");
+        }
+        if (transactionIndex >= event.getPayloadCount()) {
+            throw new IllegalArgumentException("Event does not have a transaction with index:" + transactionIndex);
```

Comment:
missing trailing space after `:` in exception string

Reply:
added